File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/actions/fight/AbstractFightDispellableEffect.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait AbstractFightDispellableEffect extends TypeDeserializer {
  val typeId = 206

  protected trait Type extends BaseType with Serializable {
    val typeId = AbstractFightDispellableEffect.typeId

    val uid: Int
    val targetId: Int
    val turnDuration: Short
    val dispelable: Byte
    val spellId: Short
    val parentBoostUid: Int

    def serialize(buf: Buffer) {
      buf.writeInt(uid)
      buf.writeInt(targetId)
      buf.writeShort(turnDuration)
      buf.writeByte(dispelable)
      buf.writeShort(spellId)
      buf.writeInt(parentBoostUid)
    }

  }

  protected def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val uid = buf.readInt()
    val targetId = buf.readInt()
    val turnDuration = buf.readShort()
    val dispelable = buf.readByte()
    val spellId = buf.readShort()
    val parentBoostUid = buf.readInt()

    create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid)
  }

}

object AbstractFightDispellableEffect extends AbstractFightDispellableEffect {
  case class TypeImpl(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int) extends Type
  override def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int) = TypeImpl(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int) = create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/AbstractFightDispellableEffect.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AbstractFightDispellableEffect extends NetworkType {
    public static final short TYPE_ID = 206;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int uid;
    public int targetId;
    public short turnDuration;
    public byte dispelable;
    public short spellId;
    public int parentBoostUid;
    
    public AbstractFightDispellableEffect() { }
    
    public AbstractFightDispellableEffect(int uid, int targetId, short turnDuration, byte dispelable, short spellId, int parentBoostUid) {
        this.uid = uid;
        this.targetId = targetId;
        this.turnDuration = turnDuration;
        this.dispelable = dispelable;
        this.spellId = spellId;
        this.parentBoostUid = parentBoostUid;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(uid);
        buf.writeInt(targetId);
        buf.writeShort(turnDuration);
        buf.writeByte(dispelable);
        buf.writeShort(spellId);
        buf.writeInt(parentBoostUid);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        uid = buf.readInt();
        if (uid < 0)
            throw new RuntimeException("Forbidden value on uid = " + uid + ", it doesn't respect the following condition : uid < 0");
        targetId = buf.readInt();
        turnDuration = buf.readShort();
        dispelable = buf.readByte();
        if (dispelable < 0)
            throw new RuntimeException("Forbidden value on dispelable = " + dispelable + ", it doesn't respect the following condition : dispelable < 0");
        spellId = buf.readShort();
        if (spellId < 0)
            throw new RuntimeException("Forbidden value on spellId = " + spellId + ", it doesn't respect the following condition : spellId < 0");
        parentBoostUid = buf.readInt();
        if (parentBoostUid < 0)
            throw new RuntimeException("Forbidden value on parentBoostUid = " + parentBoostUid + ", it doesn't respect the following condition : parentBoostUid < 0");
    }
    
}
